include/service/entities/site_service.js

```diff
@@ -0,0 +1,491 @@
+    SiteService.prototype.getSiteMap = function(cb) {
+        var self  = this;
+        var tasks = {
+
+             active: function(callback) {
+                 self.getActiveSites(callback);
+             },
+
+             inactive: function(callback) {
+                 self.getInactiveSites(callback);
+             }
+        };
+        async.series(tasks, function(err, results) {
+            cb(err, results);
+        });
+    };
+
+    /**
+     * Get site name given a unique id.
+     * @method getSiteNameByUid
+     * @param {String} uid - unique id
+     * @param {Function} cb - the callback function
+     */
+    SiteService.prototype.getSiteNameByUid = function(uid, cb) {
+        var dao = new pb.DAO();
+        dao.q(SITE_COLL, {select: pb.DAO.SELECT_ALL, where: {uid: uid} }, function(err, result) {
+            var siteName = (!uid || uid === SiteService.GLOBAL_SITE) ? 'global' : '';
+
+            if(pb.util.isError(err)) {
+                pb.log.error(err);
+            }else if(result && result.length > 0) {
+                siteName = result[0].displayName;
+            }
+
+            cb(siteName);
+        });
+    };
+
+    /**
+     * Checks to see if a proposed site display name or hostname is already in the system
+     * @method isDisplayNameOrHostnameTaken
+     * @param {String}   displayName - desired name to display
+     * @param {String}   hostname - hostname of the site
+     * @param {String}   id - Site object Id to exclude from the search
+     * @param {Function} cb - Callback function
+     */
+    SiteService.prototype.isDisplayNameOrHostnameTaken = function(displayName, hostname, id, cb) {
+        this.getExistingDisplayNameHostnameCounts(displayName, hostname, id, function(err, results) {
+
+            var result = results === null;
+            if (!result) {
+                for(var key in results) {
+                    result |= results[key] > 0;
+                }
+            }
+            cb(err, result);
+        });
+    };
+
+
+    /**
+     * Gets the total counts of a display name and hostname in the site collection
+     *
+     * @method getExistingDisplayNameHostnameCounts
+     * @param {String}   displayName - site display name
+     * @param {String}   hostname - site hostname
+     * @param {String}   id - Site object Id to exclude from the search
+     * @param {Function} cb - Callback function
+     */
+    SiteService.prototype.getExistingDisplayNameHostnameCounts = function(displayName, hostname, id, cb) {
+        if (util.isFunction(id)) {
+            cb = id;
+            id = null;
+        }
+
+        var getWhere = function(where) {
+            if (id) {
+                where[pb.DAO.getIdField()] = pb.DAO.getNotIDField(id);
+            }
+            return where;
+        };
+        var dao   = new pb.DAO();
+        var tasks = {
+            displayName: function(callback) {
+                var expStr = util.escapeRegExp(displayName.toLowerCase) + '$';
```

Comment:
You are passing a function rather than passing a display name.  I think it should be:
```
var expStr = '^' + util.escapeRegExp(displayName.toLowerCase()) + '$';
```